File: sorting/selection_sort.py
def swap(a, b):
	return b,a

def selection_sort(arr):
	for i in range(0, len(arr)):
		min = i;
		for j in range(i+1, len(arr)):
			if arr[j] < arr[min]:
				min = j
		arr[i], arr[min] = swap(arr[i], arr[min])

if __name__ == "__main__":
	arr = [10, 9, 8, 7, 6, 5, 4, 3, 2, 1]
	print(arr)
	selection_sort(arr)
	print(arr)

File: sorting/selection_sort.c
#include<stdio.h>

void swap(int *a, int *b) 
{
	int tmp;
	tmp = *a;
	*a = *b;
	*b = tmp;
}
void selection_sort(int arr[], int n)
{
	int i, j, min;
	for (i=0; i<n; i++){
		min = i;
		for (j=i+1; j<n; j++){
			if (arr[j] < arr[min]) 
				min = j;
		}
		swap(&arr[min], &arr[i]);
	}
}

int main()
{
	int arr[10] = {10, 9, 8, 7, 6, 5, 4, 3, 2, 1};
   	int i, n;
	n = sizeof(arr)/sizeof(arr[0]);
	selection_sort(arr, n);
	for(i=0; i<n; i++)
		printf("%d\t", arr[i]);
	return 0;
}
